bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -197,11 +229,38 @@ final class BagsManager(executor: ScheduledThreadPoolExecutor) {
         case _                             => // Ignore
       }
   }
+
+  /** BAGS status changes. **/
+  def bagsStatus(key: SPNodeKey): Option[BagsStatus] =
+    bagsStatuses.synchronized(bagsStatuses.get(key))
+
+  /** Listeners for BAGS status changes. **/
+  private var listeners: List[BagsStatusListener] = Nil
+
+  def addBagsStatusListener(l: BagsStatusListener): Unit =
+    if (!listeners.contains(l))
+      listeners = l :: listeners
+
+  def removeBagsStatusListener(l: BagsStatusListener): Unit =
+    listeners = listeners.filterNot(_ == l)
+
+  def clearBagsStatusListeners(): Unit =
+    listeners = Nil
+
+  private def notifyBagsStatusListeners(obs: ISPObservation, oldStatus: Option[BagsStatus], newStatus: Option[BagsStatus]): Unit = {
+    listeners.foreach(_.bagsStatusChanged(obs, oldStatus.asGeminiOpt, newStatus.asGeminiOpt))
```

Comment:
I prefer that we not use `Try` because it has some design issues, but I'm not going to make that argument right now. Whatever you do please be sure to log the exception.